test/src/databases/all/functions.spec.ts

```diff
@@ -0,0 +1,1071 @@
+  describe('pi', () => {
+    it(`is pi - ${databaseName}`, async () => {
+      await funcTest('abs(pi() - 3.141592653589793) < 0.0000000000001', true);
+    });
+  });
+
+  describe('byte_length', () => {
+    it(`works - ${databaseName}`, async () => {
+      await funcTestMultiple(
+        ["byte_length('hello')", 5],
+        ["byte_length('©')", 2],
+        ['byte_length(null)', null]
+      );
+    });
+  });
+  describe('ifnull', () => {
+    it(`works - ${databaseName}`, async () => {
+      await funcTestMultiple(
+        ["ifnull('a', 'b')", 'a'],
+        ["ifnull(null, 'b')", 'b'],
+        ["ifnull('a', null)", 'a'],
+        ['ifnull(null, null)', null]
+      );
+    });
+  });
+  describe('coalesce', () => {
+    it(`works - ${databaseName}`, async () => {
+      await funcTestMultiple(
+        ["coalesce('a')", 'a'],
+        ["coalesce('a', 'b')", 'a'],
+        ["coalesce(null, 'a', 'b')", 'a'],
+        ["coalesce(null, 'b')", 'b'],
+        ["coalesce('a', null)", 'a'],
+        ['coalesce(null, null)', null],
+        ['coalesce(null)', null]
+      );
+    });
+  });
+  describe('nullif', () => {
+    it(`works - ${databaseName}`, async () => {
+      await funcTestMultiple(
+        ["nullif('a', 'a')", null],
+        ["nullif('a', 'b')", 'a'],
+        ["nullif('a', null)", 'a'],
+        ['nullif(null, null)', null],
+        ['nullif(null, 2)', null]
+      );
+    });
+  });
+  describe('chr', () => {
+    it(`works - ${databaseName}`, async () => {
+      await funcTestMultiple(
+        ['chr(65)', 'A'],
+        ['chr(255)', 'ÿ'],
+        ['chr(null)', null],
+        // BigQuery's documentation says that `chr(0)` returns the empty string, but it doesn't,
+        // it actually returns the null character. We generate code so that it does this.
+        ['chr(0)', '']
+      );
+    });
+  });
+  describe('ascii', () => {
+    it(`works - ${databaseName}`, async () => {
+      await funcTestMultiple(
+        ["ascii('A')", 65],
+        ["ascii('ABC')", 65],
+        ["ascii('')", 0],
+        ['ascii(null)', null]
+      );
+    });
+  });
+  describe('unicode', () => {
+    it(`works - ${databaseName}`, async () => {
+      await funcTestMultiple(
+        ["unicode('A')", 65],
+        ["unicode('â')", 226],
+        ["unicode('âBC')", 226],
+        ["unicode('')", 0],
+        ['unicode(null)', null]
+      );
+    });
+  });
+
+  describe('repeat', () => {
+    it(`works - ${databaseName}`, async () => {
+      await funcTestMultiple(
+        ["repeat('foo', 0)", ''],
+        ["repeat('foo', 1)", 'foo'],
+        ["repeat('foo', 2)", 'foofoo'],
+        ['repeat(null, 2)', null],
+        ["repeat('foo', null)", null]
+      );
+    });
+    // TODO how does a user do this: the second argument needs to be an integer, but floor doesn't cast to "integer" type.
+    it.skip(`works floor decimal - ${databaseName}`, async () => {
+      await funcTest("repeat('foo', floor(2.5))", 'foofoo');
+    });
```

Comment:
There's currently no way to use `floor(some_decimal)` in a place that requires an integer, because there's no way to cast to an integer, and `floor` doesn't do it.